Sprint_2

sergkolosov opened the pull request.

Commit:
Added a fixture with autouse=True. Creates an instance of a class with three books

Commit:
test_add_new_book_add_two_books fixed for three books

Commit:
fixed @pytest.fixture collector
added @pytest.fixture favorites
fixed test_add_new_book_add_two_books changed test_add_new_book_add_three_books_count
added test_add_new_book_add_three_books_value
added test_add_new_book_add_impossible_add_book_twice
added test_set_book_rating_ten
added test_set_book_rating_impossible_set_rating_book_not_in_list
added test_set_book_rating_impossible_set_rating_book_less_than_one
added test_set_book_rating_impossible_set_rating_book_more_than_ten
added test_set_book_rating_book_that_was_not_added_has_no_rating
added test_get_books_with_specific_list_rating_five_count
added test_get_books_with_specific_list_rating_five_value
added test_add_book_in_favorites_two_books_count
added test_add_book_in_favorites_two_books_value
added test_add_book_in_favorites_cannot_add_book_is_not_in_the_dictionary
added test_delete_book_from_favorites_one_book_count
added test_delete_book_from_favorites_one_book_value

Commit:
filled in

Commit:
minor edits

sergkolosov commented on tests.py:
```diff
@@ -1,24 +1,109 @@
+
+    def test_add_new_book_add_three_books_value(self):
+        # проверяем, что добавились именно те книги и их исходный рейтинг
+        assert self.collector.get_books_rating() == {self.book_1: 1, self.book_2: 1, self.book_3: 1}
```
Согласен. Исправил.

sergkolosov commented on tests.py:
```diff
@@ -1,24 +1,109 @@
+    def test_set_book_rating_ten(self):
+        # проверяем добавление книге рейтинга 10 (граничное значение) и получение рейтинга книги по названию
+        self.collector.set_book_rating(self.book_1, 10)
+        assert self.collector.get_books_rating()[self.book_1] == 10
```
Не понял коммента. В 51 строке используется именно get_book_rating

kadirovgm replied:
Нет, у ты используешь `get_books_rating` для вывода словаря и обращаешься в элементу словаря, а у нас есть специальный метод `get_book_rating`, где как раз по названию книги выводится его рейтинг

sergkolosov replied:
Возможно речи о методе get(), для получения значения по ключу? Исправил.

sergkolosov replied:
Всё, разобрался. Спасибо!

Commit:
improved test_add_new_book_add_three_books_value
changed test_set_book_rating_ten to test_set_book_rating_boundary_values
changed test_delete_book_from_favorites_one_book_count to test_delete_book_from_favorites_one_book
improved test_add_book_in_favorites_cannot_add_book_is_not_in_the_dictionary
fixed test_set_book_rating_book_that_was_not_added_has_no_rating
fixed test_set_book_rating_impossible_set_rating_book_not_in_list
fixed test_set_book_rating_impossible_set_rating_book_less_than_one
fixed test_set_book_rating_impossible_set_rating_book_more_than_ten
redid the comments to docstring

Commit:
Fixed the list of checks

kadirovgm commented on tests.py:
```diff
@@ -5,105 +5,112 @@
-        # проверяем добавление книге рейтинга 10 (граничное значение) и получение рейтинга книги по названию
-        self.collector.set_book_rating(self.book_1, 10)
-        assert self.collector.get_books_rating()[self.book_1] == 10
+    @pytest.mark.parametrize('boundary_values', [1, 2, 9, 10])
```
Можно лучше: граничные значения скорее: 0 ->1, 1->1, 10->10, 11->1
Можно было два теста сделать для валидных и невалидных значений
